### PR TITLE
astype #6242

### PR DESCRIPTION
**Pull request by codeamic**

### CODE REVIEW

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -122,7 +121,10 @@ def searchsorted(self, v, side="left", sorter=None):
 
     def squeeze(self, axis=None):
         return np_frontend.squeeze(self.data, axis=axis)
-
+    #Add astype instance method
+    def astype(self,dtype,order="K",casting="unsafe",subok=True,copy=True):
```
Comment (zaeemansari70) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:125]: You haven't implemented the `test_function` which should be implemented to complete the task, I'll link you to the document which would help you in the process.
https://lets-unify.ai/ivy/deep_dive/ivy_frontends_tests.html#frontend-instance-method-tests
Also, there are some lint errors which also need to be fixed 🙂

**Review by zaeemansari70 — COMMENTED** [review]
Hey! 
Thank you soo much for contributing to ivy! 🙂
Your PR looks amazing ✨!
Just requested a few changes and left comments on them which needs to be addressed before we can merge your PR! 🙂
Thanks!

**Comment by codeamic** [comment]
close #6242

**Comment by codeamic** [comment]
Hi zaeemansari70 , please could you review.

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Awesome, thanks for the changes! 🙂
Just a few more changes required, I've mentioned them below!

1. In the `files changed tab`, your changes are also being showed in the `test_array_api` folder, which shouldn't be the case, changes should show only in the 2 files, `ndarray.py` and `test_ndarray.py` in this case. Kindly remove the changes in the `test_array_api` folder.
2. Also, there are multiple lint errors, you have to resolve them too.

Thanks! 🙂

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by aditya-pola** [comment]
Hey @codeamic :) apologies for the delay, I did not get notification of this reassignment.
Your code seems to be failing the lint checks. You can automatically resolve most of these be installing the [pre-commit hooks](https://lets-unify.ai/ivy/contributing/setting_up.html#pre-commit)
The changes to test_array_api are not relevant to this PR as zaeemansari70 has kindly pointed out. Please revert them.

**Pull request closed without merging**